api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/convolutional/Conv2DTranspose.kt

```diff
@@ -0,0 +1,126 @@
+ * @property [kernelSize] size of the convolutional kernel (two numbers)
+ * @property [strides] strides of the convolution for each dimension of the input tensor (four numbers)
+ * @property [dilations] dilations of the convolution for each dimension of the input tensor (four numbers).
+ *           Currently, dilation values greater than 1 are not supported on cpu.
+ * @property [activation] activation function
+ * @property [kernelInitializer] initializer for the kernel
+ * @property [biasInitializer] initializer for the bias
+ * @property [kernelRegularizer] regularizer for the kernel
+ * @property [biasRegularizer] regularizer for the bias
+ * @property [activityRegularizer] regularizer function applied to the output of the layer
+ * @property [padding] type of padding to use
+ * @property [outputPadding] the amount of explicit padding to use (eight numbers: two for each dimension).
+ * @property [useBias] a flag that specifies if the bias should be used
+ * @param [name] custom layer name
+ */
+public class Conv2DTranspose(
+    public override val filters: Int = 3,
+    public override val kernelSize: IntArray = intArrayOf(3, 3),
+    public override val strides: IntArray = intArrayOf(1, 1, 1, 1),
+    public override val dilations: IntArray = intArrayOf(1, 1, 1, 1),
+    public override val activation: Activations = Activations.Relu,
+    public override val kernelInitializer: Initializer = HeNormal(),
+    public override val biasInitializer: Initializer = HeUniform(),
+    public override val kernelRegularizer: Regularizer? = null,
+    public override val biasRegularizer: Regularizer? = null,
+    public override val activityRegularizer: Regularizer? = null,
+    public override val padding: ConvPadding = ConvPadding.SAME,
+    public override val outputPadding: IntArray? = null,
+    public override val useBias: Boolean = true,
+    name: String = ""
+) : ConvTranspose(dimensions = 2, name), NoGradients {
+
+    init {
+        requireArraySize(kernelSize, dimensions, "kernelSize")
+        requireArraySize(strides, dimensions + 2, "strides")
+        requireArraySize(dilations, dimensions + 2, "dilations")
+        if (outputPadding != null) requireArraySize(outputPadding, 2 * (dimensions + 2), "outputPadding")
+        isTrainable = false
+    }
+
+    override fun convImplementation(tf: Ops, input: Operand<Float>): Operand<Float> {
+        return tf.nn.conv2dBackpropInput(
+            tf.shapeWithDynamicBatchSize(outputShape, input),
+            kernel.variable,
+            input,
+            strides.toLongList(),
+            if (outputPadding != null) EXPLICIT else padding.paddingName,
+            *buildOptions(
+                dilations,
+                outputPadding?.withStandardPadding(
+                    padding,
+                    kernelSize,
+                    dilations
+                )
+            )
+        )
+    }
+
+    internal companion object {
```

Comment:
Methods of this companion are used in the both, Conv1DTranspose and Conv2DTranspose, probably need we need a separate place with top-level functions a la Util class.